web/elm/src/Dashboard/Grid/Constants.elm

```diff
@@ -21,17 +21,27 @@ cardBodyHeight =
 
 padding : number
 padding =
-    25
+    32
+
+
+carHeaderVerticalSpace : number
+carHeaderVerticalSpace =
+    8
 
 
 cardHeaderHeight : number -> number
 cardHeaderHeight numRows =
-    20 + 30 * numRows
+    32 + 21 * numRows + (numRows - 1) * carHeaderVerticalSpace
```

Comment:
I wonder if we really care to make the spacing between rows "exactly" 8px - the thing is, the 21px line height is already imprecise, since most characters aren't actually 21px tall. Maybe we can do a more even number instead, if for nothing else than to make the heights little more sane?

One thing I tried was removing the padding div between rows and just making the line-height `32px` - the problem there was that the top and bottom row looked weird since there was so much spacing on the top and bottom. But maybe we could absorb some of the 16px header padding into the rows - that is, there's still 16px of empty space between the edge of the header and the top of the text, but some of that is from the empty space created by the `line-height`, rather than the `padding` on the header.

```diff
diff --git a/web/elm/src/Dashboard/Grid/Constants.elm b/web/elm/src/Dashboard/Grid/Constants.elm
index 448fb1ce0..6b5387a6f 100644
--- a/web/elm/src/Dashboard/Grid/Constants.elm
+++ b/web/elm/src/Dashboard/Grid/Constants.elm
@@ -1,6 +1,9 @@
 module Dashboard.Grid.Constants exposing
     ( cardBodyHeight
     , cardHeaderHeight
+    , cardHeaderPadding
+    , cardHeaderPaddingOverlap
+    , cardHeaderRowHeight
     , cardWidth
     , groupHeaderHeight
     , padding
@@ -24,14 +27,27 @@ padding =
     32
 
 
-carHeaderVerticalSpace : number
-carHeaderVerticalSpace =
+cardHeaderRowHeight : number
+cardHeaderRowHeight =
+    32
+
+
+cardHeaderPadding : number
+cardHeaderPadding =
+    16
+
+
+cardHeaderPaddingOverlap : number
+cardHeaderPaddingOverlap =
+    -- this is the number of pixels of the padding defined in cardHeaderPadding
+    -- that cardHeaderRowHeight should contribute. this is to avoid having too
+    -- much padding on the top/bottom row
     8
 
 
 cardHeaderHeight : number -> number
 cardHeaderHeight numRows =
-    32 + 21 * numRows + (numRows - 1) * carHeaderVerticalSpace
+    2 * (cardHeaderPadding - cardHeaderPaddingOverlap) + cardHeaderRowHeight * numRows
 
 
 groupHeaderLineHeight : number
diff --git a/web/elm/src/Dashboard/Pipeline.elm b/web/elm/src/Dashboard/Pipeline.elm
index a21e683fc..626d19a85 100644
--- a/web/elm/src/Dashboard/Pipeline.elm
+++ b/web/elm/src/Dashboard/Pipeline.elm
@@ -277,12 +277,8 @@ transition =
 headerView : PipelinesSection -> Pipeline -> Bool -> Float -> Bool -> Bool -> Html Message
 headerView section pipeline resourceError headerHeight viewingInstanceGroups inInstanceGroup =
     let
-        verticalSpacer =
-            Html.div [ style "height" "8px" ] []
-
         rows =
-            List.intersperse verticalSpacer
-                (headerRows section viewingInstanceGroups pipeline inInstanceGroup)
+            headerRows section viewingInstanceGroups pipeline inInstanceGroup
 
         resourceErrorElem =
             Html.div
diff --git a/web/elm/src/Dashboard/Styles.elm b/web/elm/src/Dashboard/Styles.elm
index 59301c390..de693dd80 100644
--- a/web/elm/src/Dashboard/Styles.elm
+++ b/web/elm/src/Dashboard/Styles.elm
@@ -212,11 +212,18 @@ noPipelineCardHeader =
 
 pipelineCardHeader : Float -> List (Html.Attribute msg)
 pipelineCardHeader height =
+    let
+        paddingHorizontal =
+            String.fromFloat GridConstants.cardHeaderPadding ++ "px"
+
+        paddingVertical =
+            String.fromFloat (GridConstants.cardHeaderPadding - GridConstants.cardHeaderPaddingOverlap) ++ "px"
+    in
     [ style "background-color" Colors.card
     , style "color" Colors.dashboardPipelineHeaderText
     , style "font-size" "1.5em"
     , style "letter-spacing" "0.1em"
-    , style "padding" "16px"
+    , style "padding" <| paddingVertical ++ " " ++ paddingHorizontal
     , style "height" <| String.fromFloat height ++ "px"
     , style "box-sizing" "border-box"
     ]
@@ -230,7 +237,7 @@ instanceGroupCardHeader =
 pipelineName : List (Html.Attribute msg)
 pipelineName =
     [ style "width" "240px"
-    , style "line-height" "21px"
+    , style "line-height" <| String.fromFloat GridConstants.cardHeaderRowHeight ++ "px"
     , style "white-space" "nowrap"
     , style "overflow" "hidden"
     , style "text-overflow" "ellipsis"
```